packages/core/src/kernelCoreTemplate.js

```diff
@@ -460,5 +471,17 @@
       }
       return protoChain
     }
+
+    function shouldAvoidProp(propsToAvoid, prop) {
+      for (const avoid of propsToAvoid) {
+        if (avoid instanceof RegExp && avoid.test(prop)) {
+          return true
+        }
+        if (propsToAvoid.has(prop)) {
```

Comment:
(could) create a test covering that it's not possible to accidentally match the string representation of the regex by setting `window['/as.good.a.field.name.as.any/']`

Reply:
Not too worried about such a scenario